internal/controllers/objectdeployments/archive_reconciler.go

```diff
@@ -226,3 +218,20 @@ func intersection(a, b []objectIdentifier) (res []objectIdentifier) {
 	}
 	return
 }
+
+func (a *archiveReconciler) garbageCollectRevisions(ctx context.Context, previousObjectSets []genericObjectSet, objectDeployment objectDeploymentAccessor) error {
+	revisionLimit := defaultRevisionLimit
+	deploymentRevisionLimit := objectDeployment.GetRevisionHistoryLimit()
+	if deploymentRevisionLimit != nil {
+		revisionLimit = *deploymentRevisionLimit
+	}
+	numToDelete := len(previousObjectSets) - (int(revisionLimit))
+	for numToDelete > 0 {
+		if err := a.client.Delete(ctx, previousObjectSets[0].ClientObject()); err != nil {
+			return fmt.Errorf("failed to delete objectset: %w", err)
+		}
+		numToDelete--
+	}
```

Comment:
```suggestion
	numToDelete := len(previousObjectSets) - int(revisionLimit)
	for _, previousObjectSet := range previousObjectSets {
		if numToDelete <= 0 {
			break
		}

		if err := a.client.Delete(ctx, previousObjectSet.ClientObject()); err != nil && !errors.IsNotFound(err) {
			return fmt.Errorf("failed to delete objectset: %w", err)
		}
		numToDelete--
	}
```

- The () arround the int-cast are not needed
- The for loop needs to delete _all_ Archived ObjectSets that are above the history limit